packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -211,6 +213,38 @@ sealed class _DebugSemanticsRoleChecks {
     node.visitChildren(validateRadioGroupChildren);
     return error;
   }
+
+  static FlutterError? _semanticsMenu(SemanticsNode node) {
+    if (node.childrenCount < 1) {
+      return FlutterError('a menu cannot be empty');
+    }
+
+    return null;
+  }
+
+  static FlutterError? _semanticsMenuBar(SemanticsNode node) {
+    if (node.childrenCount < 1) {
+      return FlutterError('a menu bar cannot be empty');
+    }
+
+    return null;
+  }
+
+  static FlutterError? _semanticsMenuItemCheckbox(SemanticsNode node) {
+    if (!node.hasFlag(SemanticsFlag.hasCheckedState)) {
+      return FlutterError('a menu item checkbox must have be checkable');
+    }
```

Comment:
also for menu item, better check the parent is a menu

Reply:
While the direct parent may not be a menu, can we check the entire ancestor chain contains the menu?

Reply:
Done!